plugins/obs-ffmpeg/obs-ffmpeg-output.c

```diff
@@ -1040,13 +1040,13 @@ static int process_packet(struct ffmpeg_output *output)
 	output->total_bytes += packet->size;
 
 	ret = av_interleaved_write_frame(output->ff_data.output, packet);
-	if (ret < 0) {
-		av_packet_free(&packet);
```

Comment:
I'm just checking visually but why this line is removed?

Reply:
reference ffmpeg demo how to using  **av_interleaved_write_frame** api

```
// doc\examples\remuxing.c 
int main(int argc, char **argv)
{
     ...
     ret = av_interleaved_write_frame(ofmt_ctx, &pkt);
        if (ret < 0) {
            fprintf(stderr, "Error muxing packet\n");
            break;
        }
        av_packet_unref(&pkt);

     ...
}
```

Reply:
In that example, the code finally reaches `av_packet_free(&pkt);`
https://github.com/FFmpeg/FFmpeg/blob/48be6616d0536c5b0ff3ee58caee4c024ca64116/doc/examples/remuxing.c#L180-L181

In `obs-ffmpeg-output.c`, the if statement has `return` so that the packet should be freed.

I tested with injecting error like below and current code handling the error of `av_interleaved_write_frame` looks ok.
```diff
diff --git a/plugins/obs-ffmpeg/obs-ffmpeg-output.c b/plugins/obs-ffmpeg/obs-ffmpeg-output.c
index 7d28ad669..67387fdb8 100644
--- a/plugins/obs-ffmpeg/obs-ffmpeg-output.c
+++ b/plugins/obs-ffmpeg/obs-ffmpeg-output.c
@@ -1039,6 +1039,7 @@ static int process_packet(struct ffmpeg_output *output)
 
        output->total_bytes += packet->size;
 
+       packet->stream_index = 11;
        ret = av_interleaved_write_frame(output->ff_data.output, packet);
        if (ret < 0) {
                av_packet_free(&packet);
```